server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -246,10 +250,19 @@ public static void deleteTable(TableId tableId, boolean insertDeletes, ServerCon
 
   public static void removeUnusedWALEntries(ServerContext context, KeyExtent extent,
       final Collection<LogEntry> entries, ServiceLock zooLock) {
-    TabletMutator tablet = context.getAmple().mutateTablet(extent);
-    entries.forEach(tablet::deleteWal);
-    tablet.putZooLock(context.getZooKeeperRoot(), zooLock);
-    tablet.mutate();
+    try (ConditionalTabletsMutator mutator = context.getAmple().conditionallyMutateTablets()) {
+      ConditionalTabletMutator mut = mutator.mutateTablet(extent).requireAbsentOperation()
```

Comment:
I think you need to add a condition here to require for the conditional mutation. The condition you have is only set for the rejection handler. The rejection handler I believe is only used to check if it should succeed if it was reported as rejected but not check if it was successful. So I think you need something like the following which would require absent logs and then you'd need to implement requireLock()

```java
      ConditionalTabletMutator mut = mutator.mutateTablet(extent).requireAbsentOperation()
          .requireAbsentLogs().requireLock(zooLock)
          .putZooLock(context.getZooKeeperRoot(), zooLock);
```

Reply:
Edit: I meant to say in my previous post before I modified it that to add that it should require NOT having the current zooLock since obviously it's setting the current one, but that may not matter. I just noticed in the rejection handler you were checking that the zooLock set equals the current lock, so I was thinking we should require there not be a lock already when submitting but this might not be necessary

Reply:
Given my comment above, I don't see anything in the code that removes the lock column from the metadata. So, adding something like `requireAbsentLock` may not work.